composeApp/src/commonMain/kotlin/sp/bvantur/tasky/event/data/CreateEventRepositoryImpl.kt

```diff
@@ -0,0 +1,30 @@
+package sp.bvantur.tasky.event.data
+
+import sp.bvantur.tasky.event.domain.CreateEventRepository
+import sp.bvantur.tasky.event.domain.model.Attendee
+
+class CreateEventRepositoryImpl(private val remoteDataSource: CreateEventRemoteDataSource) : CreateEventRepository {
+    override suspend fun getAttendee(email: String): Result<Attendee?> {
+        val response = remoteDataSource.getAttendee(email)
+        val exception = response.exceptionOrNull()
+        if (exception != null) {
```

Comment:
What do you think, it is ok if we check for a failure state like that in our code?

Reply:
I'm not a fan of this Kotlin Result class with this architecture, since it quickly couples your domain layer to HTTP related exceptions and therefore your HTTP library. Also, it forces you to work with these exceptions across the app. You can't easily rely on using the Exception's message to show an error, since that isn't localized by default.

I'm always using this Result wrapper which works with enums as error codes instead. This doesn't have that issue:
```
sealed interface Result<out D, out E: Error> {
    data class Success<out D>(val data: D): Result<D, Nothing>
    data class Error<out E: com.plcoding.tasky.core.domain.util.Error>(val error: E): Result<Nothing, E>
}

inline fun <T, E: Error, R> Result<T, E>.map(map: (T) -> R): Result<R, E> {
    return when(this) {
        is Result.Error -> Result.Error(error)
        is Result.Success -> Result.Success(map(data))
    }
}

fun <T, E: Error> Result<T, E>.asEmptyDataResult(): EmptyResult<E> {
    return map {  }
}

inline fun <T, E: Error> Result<T, E>.onSuccess(action: (T) -> Unit): Result<T, E> {
    return when(this) {
        is Result.Error -> this
        is Result.Success -> {
            action(data)
            this
        }
    }
}
inline fun <T, E: Error> Result<T, E>.onError(action: (E) -> Unit): Result<T, E> {
    return when(this) {
        is Result.Error -> {
            action(error)
            this
        }
        is Result.Success -> this
    }
}

typealias EmptyResult<E> = Result<Unit, E>
```
It handles data and errors in a type-safe way. You'll need a marker interface to mark error enums:
```
interface Error
```
And then you can extend this across the codebase with different error types, e.g.:
```
sealed interface DataError: Error {
    enum class Remote: DataError {
        NOT_FOUND,
        NO_INTERNET,
        UNAUTHORIZED,
        ...
    }
    
    enum class Local: DataError {
        DISK_FULL,
        ...
    }
}
```
Works great:
```
repository.signUp(payload).onSuccess { responseData ->

}.onError { error -> 
    // toUiText() can be an extension of DataError which maps an error to the corresponding message
    _state.update { it.copy(error = error.toUiText()) }
}
```